crates/router/src/core/payment_methods/cards.rs

```diff
@@ -253,6 +254,83 @@ pub async fn add_card_hs(
     customer_id: String,
     merchant_account: &domain::MerchantAccount,
 ) -> errors::CustomResult<(api::PaymentMethodResponse, bool), errors::VaultError> {
+    let store_card_payload =
+        call_to_card_hs(state, &card, None, &customer_id, merchant_account).await?;
+
+    let payment_method_resp = payment_methods::mk_add_card_response_hs(
+        card,
+        store_card_payload.card_reference,
+        req,
+        &merchant_account.merchant_id,
+    );
+    Ok((
+        payment_method_resp,
+        store_card_payload.duplicate.unwrap_or(false),
+    ))
+}
+
+#[instrument(skip_all)]
+pub async fn get_payment_method_from_hs_locker<'a>(
+    state: &'a routes::AppState,
+    key_store: &domain::MerchantKeyStore,
+    customer_id: &str,
+    merchant_id: &str,
+    payment_method_reference: &'a str,
+) -> errors::CustomResult<Secret<String>, errors::VaultError> {
+    let locker = &state.conf.locker;
+    #[cfg(not(feature = "kms"))]
+    let jwekey = &state.conf.jwekey;
+    #[cfg(feature = "kms")]
+    let jwekey = &state.kms_secrets;
+
+    let request = payment_methods::mk_get_card_request_hs(
```

Comment:
As this request is only necessary for the locker, we can skip making request for mock locker.So, can you move this request inside `!locker.mock_locker` if condition